core/src/com/unciv/ui/worldscreen/WorldMapHolder.kt

```diff
@@ -754,4 +754,37 @@ class WorldMapHolder(
     override fun draw(batch: Batch?, parentAlpha: Float) = super.draw(batch, parentAlpha)
     override fun act(delta: Float) = super.act(delta)
     override fun clear() = super.clear()
+
+    override fun checkX(deltaX: Float): Float {
+        val exploredRegion = worldScreen.viewingCiv.exploredRegion
+        var result = scrollX - deltaX
+        if(!exploredRegion.isEmpty() && exploredRegion["endVector"]!!.x - exploredRegion["startVector"]!!.x != 1f) {
+            val mapCenterX = maxX * 0.5f
+            var startX = mapCenterX+exploredRegion["xWorldVector"]!!.x
+            var endX = mapCenterX+exploredRegion["xWorldVector"]!!.y
+
+            if (startX > maxX) startX = 10f
+            if (endX < 0f) endX = maxX - 10f
+
+            if ((startX < endX && result > endX) || (endX < startX && result < startX && result > endX) && deltaX < 0)
+                result = endX
+            else if ((startX < endX && result < startX) || (endX < startX && result < startX && result > endX) && deltaX > 0)
+                result = startX
+        }
+        return result
+    }
+
+    override fun checkY(deltaY: Float): Float {
+        val exploredRegion = worldScreen.viewingCiv.exploredRegion
+        var result = scrollY + deltaY
+        if(!exploredRegion.isEmpty()) {
+            val mapCenterY = maxY * 0.5f
+            val startY = mapCenterY-exploredRegion["yWorldVector"]!!.x
```

Comment:
This is another confusing thing. You're using x,y in the vector for usages that are not, in fact, x and y.
This would be better served by topright/bottomleft vectors for world, OR just use maxWorldY, minWorldY etc in the exploredRegion object